pkg/admin/prerun/prerun.go

```diff
@@ -31,36 +32,49 @@ func (hi *HealthInfo) IsHealthy() bool {
 	return hi.Health == "healthy"
 }
 
-func Perform() error {
+type PreRun struct {
+	dataManager data.Manager
+}
+
+func New(dataManager data.Manager) *PreRun {
+	return &PreRun{
+		dataManager: dataManager,
+	}
+}
+
+func (pr *PreRun) Perform() error {
 	health, err := getHealthInfo()
 	if err != nil {
 		if errors.Is(err, errHealthFileDoesNotExist) {
 			klog.InfoS("Health file does not exist - skipping backup")
 			return nil
 		}
-		klog.ErrorS(err, "Failed to load health from disk")
 		return err
 	}
-	klog.InfoS("Loaded health info from the disk", "health", health)
 
 	if isCurr, err := containsCurrentBootID(health.BootID); err != nil {
 		return err
 	} else if isCurr {
-		klog.InfoS("Health file contains current boot - skipping backup")
+		// This might happen if microshift is (re)started after greenboot finishes running.
+		// Green script will overwrite the health.json with
+		// current boot's ID, deployment ID, and health.
+		klog.InfoS("Health file contains current boot - skipping pre-run")
 		return nil
 	}
 
-	if !health.IsHealthy() {
-		klog.InfoS("System was not healthy - skipping backup")
-		return nil
-	}
+	klog.InfoS("Previous boot", "health", health.Health, "deploymentID", health.DeploymentID, "bootID", health.BootID)
 
-	dataManager, err := data.NewManager(config.BackupsDir)
-	if err != nil {
-		return err
+	if health.IsHealthy() {
+		return pr.backup(health)
```

Comment:
As implementation grows, I expect this to not be end of the workflow.
For example, when rolling back from healthy system, we want to take backup, but then we want to restore another (compatible with rollback's deployID). Storage migration is another example.
We'll get there